### PR TITLE
Backwards compatible support for searching multiple S3 paths for plugins

### PR DESCRIPTION
**Pull request by tcorbettclark**

### CODE REVIEW

File: encapsia_cli/plugins.py [hunk 1]
```diff
@@ -115,30 +119,36 @@ class PluginInfo:
...
+        return m.group(1), m.group(2)  # (name, version)
 
     @classmethod
-    def make_from_s3_path(cls, bucket, filename):
-        pi = cls.make_from_filename(filename)
-        pi.bucket = bucket
-        return pi
+    def make_from_filename(cls, filename):
+        name, version = cls.get_name_version_from_filename(filename)
+        return cls.make_from_name_version(name, version)
```
Comment (petrem) [inline encapsia_cli/plugins.py:145]: In the other methods, you're calling the constructor. Any particular reason you're not calling `PluginInfo(None, None, name, version) here, too?

**Review by petrem — APPROVED** [review]

File: encapsia_cli/encapsia.py [hunk 2]
```diff
@@ -41,8 +41,9 @@
 # Name of local directory used to store plugins.
 local_dir = "~/.encapsia/plugins"
 
-# List of AWS S3 buckets in which to search for plugins.
-s3_buckets = ["ice-plugins"]
+# List of AWS S3 buckets (with optional paths) in which to search for plugins.
+# The ice-plugins bucket should be considered legacy, and will be removed.
+s3_buckets = ["encapsia-plugins/cmedtech/pd/staging/apps", "encapsia-plugins/cmedtech/pd/staging/middleware", "ice-plugins"]
```
Comment (zawasp) [inline encapsia_cli/encapsia.py:46]: Shouldn't the tool also look for Insights in the `pd/staging/insights` folder?

Comment (tcorbettclark) [reply encapsia_cli/encapsia.py:46]: Yes probably. I'd also like to remove the `ice-plugins`, but it is too soon for that. I'm thinking of these as examples only.

Anyway, you are right - I'll add insights too!

File: encapsia_cli/plugins.py (hunk 1 above)
Comment (tcorbettclark) [reply encapsia_cli/plugins.py:145]: Abstraction vs explicit.

I agree: explicit wins. Changing...

**Comment by tcorbettclark** [comment]
Hi @petrem and @zawasp 

Changes made as suggested. Thank you. Any further comments before I merge?

**Review by zawasp — APPROVED** [review]

**Comment by petrem** [comment]
> Hi @petrem and @zawasp
> 
> Changes made as suggested. Thank you. Any further comments before I merge?

No comments.

**Pull request merged**